src/tbb/governor.h

```diff
@@ -148,6 +155,70 @@ class governor {
         return false;
 #endif
     }
+
+#if __TBB_STATISTICS
+    enum class statistics : std::size_t {
+        executed = 0,
+        try_get,
+        got,
+        try_steal,
+        stolen,
+        empty_vic_pool,
+        size,
+    };
+    using array_stat = std::array<std::uint32_t, static_cast<std::size_t>(statistics::size)>;
+
+    class collect_statistics : no_copy {
+    public:
+        collect_statistics() = default;
+        ~collect_statistics() {
+            print_statistics();
+        }
+        void emplace(array_stat thread_stat) {
+            std::lock_guard<std::mutex> lk {m_};
+            accumulate_statistics_.emplace_back(thread_stat);
+        }
+
+        // thread-unsafe method
+        void print_statistics() {
+            int offset = 2;
+            /* std::string not so bad because of Small String Optimization */
+            const static std::string column_names[] = {
+                "executed", "try get",
+                "got", "try steal", "stolen",
+                "empty vic pool",
+            };
+            for (auto& el : column_names) {
+                std::cout << std::left << std::setw(el.size() + offset) << el;
+            }
+            std::cout << std::endl;
+            array_stat summary{};
+            for (const auto &thread_stat : accumulate_statistics) {
+                for (std::size_t k = 0; k < thread_stat.size(); ++k) {
+#if !__TBB_STATISTICS_SUM
+                    std::cout << std::left << std::setw(column_names[k].size() + offset) << thread_stat[k];
+#endif
+                    summary[k] += thread_stat[k];
+                }
+#if !__TBB_STATISTICS_SUM
+                std::cout << std::endl;
+#endif
+            }
+#if !__TBB_STATISTICS_SUM
+            std::cout << "SUMMARY:" << std::endl;
+#endif
+            for (std::size_t k = 0; k < summary.size(); ++k) {
+                std::cout << std::left << std::setw(column_names[k].size() + offset) << summary[k];
+            }
+            std::cout << std::endl;
+        }
+    private:
+        std::mutex m_;
+        std::vector<array_stat> accumulate_statistics_ {};
```

Comment:
Usually, we don't use underscore suffix to mark the member as private. Please consider different naming. I would propose `statistics_collection` or `statistics_container`.